### ISSUE TITLE
Question: Thoughts on releases?

### ISSUE THREAD

**Issue by Ben10do** [rednex/rgbds#138]
Any thoughts on when we should make a new release? The last release was all the way back in July 2015, and there've been 73 commits since then…

Granted, I'm sure lots of people will just directly use the latest version from GitHub, but package managers like Homebrew point to a release. The README also still suggests that Windows users use the latest release.

Judging that some projects were (and for others, likely are) still relying on the old linker's behaviour in 7b8d4de35c2dc284501218982fb1a05f6a1a8321, it'd probably be sensible to create a release for that commit now, and then decide at what point we should create a new release for the changes since then.

**Comment by AntonioND** [comment]
I've created this: https://github.com/rednex/rgbds/releases/tag/v0.2.5

I can't compile windows binaries right now, if someone wants to do them, I'll be happy to edit the release and add them.

**Comment by Sanqui** [comment]
Given that the release is not backwards compatible, it should not to be minor.  Can you please bump it up to v0.3.0?

**Comment by Sanqui** [comment]
Alternatively, we could full on switch to (semantic versioning)[http://semver.org/] and release v1.0.0, but I understand the reluctance.

**Comment by Ben10do** [comment]
I assume you mean this for the next release? I don't recall any backwards-incompatible changes at 7b8d4de35c2dc284501218982fb1a05f6a1a8321 (v0.2.5), only afterwards.

**Comment by Sanqui** [comment]
Oh, I see, my mistake.  Apologies!

**Comment by Ben10do** [comment]
No worries! It's still important to think about for future releases 😉

**Comment by AntonioND** [comment]
Actually...

> 4. Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.

:P

**Comment by Ben10do** [comment]
I've blown the cobwebs off a Windows 10 VM, and have tried to compile rgbds for it. I've hardly tested it extensively, but it seems to be working, and doesn't appear to require that Cygwin is installed…

[rgbds-0.2.5-win64.zip](https://github.com/rednex/rgbds/files/888575/rgbds-0.2.5-win64.zip)

(Compiled for 64-bit because it's 2017, but I'm sure I can compile a 32-bit version if that's a massive problem)

**Comment by AntonioND** [comment]
I've updated the release with your binaries. https://github.com/rednex/rgbds/releases/tag/v0.2.5

**Comment by AntonioND** [comment]
I'll add a note to the release info, then. I don't think it's a good idea to add a common dll to the folder in case the user has a more updated version in the PC. However, I'd like to hear your thoughts, @chickendude as I'm not the one that is going to use the Windows build.

**Comment by AntonioND** [comment]
While I'm at it, I intend to release version 0.3.0 over the following weeks. First, I want to do a bit of cleanup.

**Comment by Ben10do** [comment]
I'm glad to hear that it works, @chickendude! I think rgbgfx was the one I forgot to check, hence the libpng dependency…

@AntonioND: Apparently I compiled it with libpng-devel 1.6.28-1. I like the idea of statically compiling libpng if possible, but again, I'm not going to be the one using these binaries!

**Comment by AntonioND** [comment]
I've tried to compile this in Windows with MinGW, and it will as long as the `-Werr` option is removed. Well, that, and I couldn't compile rgbgfx because of libpng and pkg-config.

```
src/asm/lexer.c: In function 'lex_AddStrings':
src/asm/lexer.c:349:14: warning: implicit declaration of function 'strdup' [-Wimplicit-function-declaration]
     (char *) strdup(lex->tzName)) != NULL) {
              ^
src/asm/symbol.c: In function 'sym_AddNewMacroArg':
src/asm/symbol.c:437:22: warning: implicit declaration of function 'strdup' [-Wimplicit-function-declaration]
    newmacroargs[i] = strdup(s);
                      ^

src/link/lexer.c: In function 'yy_init_buffer':
src/link/lexer.c:1448:48: warning: implicit declaration of function 'fileno' [-Wimplicit-function-declaration]
         b->yy_is_interactive = file ? (isatty( fileno(file) ) > 0) : 0;
                                                ^
```

I guess that the best option for me would be to cross compile from Linux.

**Comment by Ben10do** [comment]
I used Cygwin and MinGW64 together, which compiled without problems (after specifying `CC` and `YACC`). It's not the Cygwin DLL that's the issue, but the libpng DLL, but nevertheless, I've not managed to statically compile it yet.

**Comment by chickendude** [comment]
@AntonioND i can't really say which would be best either as i mostly work on Linux (where it was simple enough to compile everything). I've been using some other Windows tools (NO$GMB and Pyxel Edit) which is why i tried the Windows binaries out. I think a note about rgbgfx should be enough, though.

**Comment by AntonioND** [comment]
I've released v0.3.0: https://github.com/rednex/rgbds/releases/tag/v0.3.0

It's been quite a pain to generate binaries for Windows, so I'll just go with what I managed to get working.

The next versions will be released more frequently.

I don't think it makes sense to keep this open now.
